app/services/puppet_class_importer.rb

```diff
@@ -94,6 +107,29 @@ def updated_classes_for(environment)
     ]
   end
 
+  # Gives back the classes ignored for a given environment
+  #
+  # Params:
+  #  * ++environment++:: {String} name of the environment
+  #
+  def ignored_classes_for(environment)
+    classes = proxy.classes(environment)
```

Comment:
Can you use the existing query instead of making a second request to the smart proxy? The performance of this API in some configurations may be poor so it shouldn't be called unnecessarily.

It looks like the information is broadly available via `actual_classes`, but it will need a little refactoring to keep the ignored classes.